### PR TITLE
Implemented ViewModel and Form Validated Using it

### PR DESCRIPTION
**Pull request by repo-son**
Created ViewModel Class and I have completed Binding all the fields in the form.
After that validated using ViewModel not by the Code-Behind.

### CODE REVIEW

File: TestLogin/TestLogin/TestLogin/DisplayMessages.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    internal interface DisplayMessages
+    {
+        Task ShowAsync(string message);
+    }
+    public class ServiceMessage : DisplayMessages
+    {
+        public async Task ShowAsync(string message)
+        {
+            await App.Current.MainPage.DisplayAlert("YourApp", message, "Ok");
+            throw new NotImplementedException();
```
Comment (madmadusanka) [inline TestLogin/TestLogin/TestLogin/DisplayMessages.cs:17]: Is this Tested? Why u throw a exception?

File: TestLogin/TestLogin/TestLogin/MainViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+using Xamarin.Forms;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace TestLogin
+{
+    internal class MainViewModel : INotifyPropertyChanged
+    {
+       
+        #region
```
Comment (madmadusanka) [inline TestLogin/TestLogin/TestLogin/MainViewModel.cs]: Its better if u add a  region region name ex- #region Properties

File: TestLogin/TestLogin/TestLogin/MainViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        {
+            get
+            {
+                return _password;
+            }
+            set
+            {
+                _password = value;
+                OnPropertyChanged("Password");
+                OnPropertyChanged(nameof(Password));
```
Comment (madmadusanka) [inline TestLogin/TestLogin/TestLogin/MainViewModel.cs:48]: Do we need to invoke the event for all the properties? or changed property only?

File: TestLogin/TestLogin/TestLogin/MainViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+                OnPropertyChanged(nameof(Status));
+
+            }
+
+        }
+        public Command Login
```
Comment (madmadusanka) [inline TestLogin/TestLogin/TestLogin/MainViewModel.cs]: It's better if u rename this like LoginCommand, So we can easily identify its a command

File: TestLogin/TestLogin/TestLogin/MainViewModel.cs [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+                }
+                else if (_username == "lahiru" && _password == "lahiru")
+                {
+                    await Application.Current.MainPage.DisplayAlert("Successfull", "Welcome ", "Ok");
+                    App.Current.MainPage = new MainMenu();
+                }
+                else
+                {
+                    await Application.Current.MainPage.DisplayAlert("Access Denied", "Please Check your Connection ", "Ok");
+                }
```
Comment (madmadusanka) [inline TestLogin/TestLogin/TestLogin/MainViewModel.cs:87]: "Ok" its repeated, add these strings to a constant to reuse purpose

File: TestLogin/TestLogin/TestLogin/DisplayMessages.cs (hunk 1 above)
Comment (repo-son) [reply TestLogin/TestLogin/TestLogin/DisplayMessages.cs:17]: Sorry. I forgot to Delete DisplayMessages.cs It is not serving a purpose.

File: TestLogin/TestLogin/TestLogin/MainViewModel.cs (hunk 2 above)
Comment (repo-son) [reply TestLogin/TestLogin/TestLogin/MainViewModel.cs]: Noted

File: TestLogin/TestLogin/TestLogin/MainViewModel.cs (hunk 3 above)
Comment (repo-son) [reply TestLogin/TestLogin/TestLogin/MainViewModel.cs:48]: Changed Property only

File: TestLogin/TestLogin/TestLogin/MainViewModel.cs (hunk 4 above)
Comment (repo-son) [reply TestLogin/TestLogin/TestLogin/MainViewModel.cs]: Sure. Noted

File: TestLogin/TestLogin/TestLogin/MainViewModel.cs (hunk 5 above)
Comment (repo-son) [reply TestLogin/TestLogin/TestLogin/MainViewModel.cs:87]: Ok mr.Dilan

Comment (repo-son) [reply TestLogin/TestLogin/TestLogin/MainViewModel.cs:87]: Added Alerts Class

File: TestLogin/TestLogin/TestLogin/DisplayMessages.cs (hunk 1 above)
Comment (repo-son) [reply TestLogin/TestLogin/TestLogin/DisplayMessages.cs:17]: Class Removed

File: TestLogin/TestLogin/TestLogin/MainViewModel.cs (hunk 2 above)
Comment (repo-son) [reply TestLogin/TestLogin/TestLogin/MainViewModel.cs]: Regions Properly Named

File: TestLogin/TestLogin/TestLogin/MainViewModel.cs (hunk 4 above)
Comment (repo-son) [reply TestLogin/TestLogin/TestLogin/MainViewModel.cs]: Changed to LoginCommand

**Comment by repo-son** [comment]
All issues have been solved

**Pull request merged**